app/controllers/application_controller.rb

```diff
@@ -26,6 +27,25 @@ def authorize(record, query=nil, user=nil)
     end
   end
 
+  def param_date(sym)
+    begin
+      Date.strptime(params[sym].to_s)
+    rescue
+      nil
+    end
+  end
+
+  def return_to
+    if params[:return_to]
+      return_to = params.require(:return_to)
+      proper_sig = self.make_return_to(return_to.require(:name),
+                                       return_to.require(:path))[:sig]
+      if return_to.require(:sig) == proper_sig
+        return_to.permit([:path, :name])
+      end
+    end
+  end
```

Comment:
`session` variables hang around until explicitly removed, which is not desirable here. We can definitely replace `session[:return_to]` with this GET-based version, though.